test/k8sT/Nightly.go

```diff
@@ -87,35 +90,33 @@ var _ = Describe("NightlyEpsMeasurement", func() {
 		res.ExpectSuccess(res.GetDebugMessage())
 
 		waitForPodsTime := b.Time("Wait for pods", func() {
-			pods, err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", 300)
+			pods, err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", endpointsTimeout)
+			Expect(err).Should(BeNil(),
+				"Cannot retrieve %d pods in %d seconds", endpointCount, endpointsTimeout)
 			Expect(pods).Should(BeTrue())
-			Expect(err).Should(BeNil())
 		})
-		log.WithFields(logrus.Fields{"pod creation time": waitForPodsTime}).Info("")
-
-		ciliumPod, err := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
-		Expect(err).Should(BeNil())
 
-		ciliumPod2, err := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s2)
-		Expect(err).Should(BeNil())
+		log.WithFields(logrus.Fields{"pod creation time": waitForPodsTime}).Info("")
 
-		pods := []string{ciliumPod, ciliumPod2}
+		ciliumPods, err := kubectl.GetCiliumPods(helpers.KubeSystemNamespace)
+		Expect(err).To(BeNil(), "Cannot retrieve cilium pods")
 
 		runtime := b.Time("Endpoint creation", func() {
-
 			Eventually(func() bool {
 				count := 0
-				for _, pod := range pods {
-					output := kubectl.CiliumExec(pod, "cilium endpoint list").Output().String()
-					count += strings.Count(output, "ready")
+				for _, pod := range ciliumPods {
+					status := kubectl.CiliumEndpointsStatus(pod)
+					result := map[string]int{}
+					for _, state := range status {
+						result[state]++
+					}
+					count += result[desiredState]
+					log.WithField("status", result).Debugf("Cilium %s endpoint status are:", pod)
```

Comment:
Instead of having the log message contain the pod name, add the pod name to `log.WithFields` as a field.

Reply:
Done